Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
>You may also have noticed a common fundamental thing between those sources.
They all think more like human beings, less like computers.

Yeah, they're what maintain and write the code; that's kind of how the relationship works.

> It is fact that debugging this code is easier by simply changing the type of i, in this form.

You can obsolete the need to change it by using [auto](http://en.cppreference.com/w/cpp/language/auto) in many circumstances where this would actually matter in practice, among other things (if modification of loop variables is frequent, outside of circumstances where the design as a whole is changed, either local or in another translation unit that necessitates the change, there's a clear design issue that isn't being thought out). You don't need to concern yourself with the type.

> If you had asked me for a reason why there's anything to lose from doing C99's for (int i = ... as you stated, and I replied by re-iterating that it is deprecated (assuming some crazy, hypothetical C89-compatible standard chose to deprecate it, whether you personally liked it or not), would you not find that reasoning to be recursive?

If you replied it was deprecated, and it was indeed the case (I would look for sources first to back it up or ask you to provide it), I would move code over to the non-deprecated form once confirmed; basic software maintenance 101.

> I don't believe for a second that for (size_t i = ... is any less ambiguous or open to multiple possibilities of interpretation (or even personal language implementation) than just declaring the variable first.

But it is and can be. Scoping rules exist. If they didn't, I would agree.

>They are more interested in the future because they have the power to control it. You might find this theory is paranoid, but it's really just human nature. To them, portability means little if it's only because of backwards compatibility against something they would rather have their own way.

literally_any_newer_language_spec.txt. 

> Besides, I can take a page from your book and propose that not using the register keyword opens up part of your code to accidentally "taking the address" of a variable... In ANSI C

[Sure, but this isn't a purely C codebase, so it's not enforced in this context](http://ideone.com/Wqs3lN).